src/ol_concourse/pipelines/open_edx/edx_platform_v2/grouped_docker_packer_pulumi_pipeline.py

```diff
@@ -0,0 +1,305 @@
+import sys
+
+from bridge.settings.openedx.types import OpenEdxSupportedRelease
+from ol_concourse.lib.containers import container_build_task
+from ol_concourse.lib.models.fragment import PipelineFragment
+from ol_concourse.lib.models.pipeline import (
+    AnonymousResource,
+    Command,
+    GetStep,
+    GroupConfig,
+    Identifier,
+    InParallelStep,
+    Input,
+    Job,
+    Output,
+    Pipeline,
+    Platform,
+    PutStep,
+    TaskConfig,
+    TaskStep,
+)
+from ol_concourse.lib.resources import git_repo, registry_image
+from ol_concourse.lib.jobs.infrastructure import packer_jobs, pulumi_jobs_chain
+from ol_concourse.pipelines.constants import PULUMI_WATCHED_PATHS, PULUMI_CODE_PATH
+
+from bridge.settings.openedx.accessors import (
+    filter_deployments_by_release,
+    fetch_application_version,
+)
+from bridge.settings.openedx.types import (
+    OpenEdxApplication,
+)
+
+
+def build_edx_pipeline(release_names: list[str]) -> Pipeline:
+    edx_docker_code = git_repo(
+        name=Identifier("ol-infrastructure-docker"),
+        uri="https://github.com/mitodl/ol-infrastructure",
+        # TODO MD 20230512 Fix to main branch once testing completed
+        branch="md/edxapp_docker_migration",
+        paths=[
+            "dockerfiles/openedx-edxapp/",
+            "src/ol_concourse/pipelines/open_edx/edx_platform_v2",
+        ],
+    )
+
+    container_fragments = []
+    packer_fragments = []
+    pulumi_fragments = []
+    group_configs = []
+    for release_name in releases:
+        job_names = []
+        for deployment in filter_deployments_by_release(release_name):
+            theme_git_resources = []
+            theme_get_steps = []
+
+            theme = fetch_application_version(
+                release_name, deployment.deployment_name, OpenEdxApplication.theme
+            )
+            theme_git_resource = git_repo(
+                name=Identifier(f"{release_name}-{deployment.deployment_name}-theme"),
+                uri=theme.git_origin,
+                branch=theme.release_branch,
+            )
+            theme_git_resources.append(theme_git_resource)
+            theme_get_steps.append(GetStep(get=theme_git_resource.name, trigger=False))
+
+            edx_platform_git_resources = []
+            edx_platform_get_steps = []
+            edx_platform = fetch_application_version(
+                release_name, deployment.deployment_name, OpenEdxApplication.edxapp
+            )
+            edx_platform_git_resource = git_repo(
+                name=Identifier(
+                    f"{release_name}-{deployment.deployment_name}-edx-platform"
+                ),
+                uri=edx_platform.git_origin,
+                branch=edx_platform.release_branch,
+            )
+            edx_platform_git_resources.append(edx_platform_git_resource)
+            edx_platform_get_steps.append(
+                GetStep(get=edx_platform_git_resource.name, trigger=False)
```

Comment:
We want changes in the edX code to trigger new builds so that our CI stages are always up to date

Reply:
yes good catch. this should have had a TODO before it. The automatic running was interfering with stuff so it was turned off.

Reply:
So for mitxonline --- this causes A LOT of builds which is maybe the intention but it is something to think about. I'm going to set the max-in-flight to one because they are pummeling each other at the moment.